packages/api/src/controllers/webhook.ts

```diff
@@ -241,28 +241,57 @@ app.post(
   async (req, res) => {
     const { id } = req.params;
     const webhook = await db.webhook.get(id);
-    if (!webhook) {
-      return res.status(404).json({ errors: ["not found"] });
+    if (!webhook || webhook.deleted) {
+      return res.status(404).json({ errors: ["webhook not found or deleted"] });
     }
 
-    const doc = req.body.status;
-    const webhookResponse = req.body.response;
+    const { response, errorMessage } = req.body as WebhookStatusPayload;
 
-    if (!doc) {
-      return res.status(422).json({ errors: ["missing status in payload"] });
+    if (!response || !response.response) {
+      return res.status(422).json({ errors: ["missing response in payload"] });
+    }
+    if (!response.response.body && response.response.body !== "") {
+      return res
+        .status(400)
+        .json({ errors: ["missing body in payload response"] });
     }
```

Comment:
The response should be an optional field since the webhook call could have gotten an error (`errorMessage`) but not an actual HTTP response.

Apart from that, even if we wanted to have a required field here, this logic should be handled on the schema level, by making the field required with `required: [response]`. There are some more "semantic" checks that can't be declared too easily in a JSON schema, but otherwise we should not be doing "json schema" checks in our code as much as possible.